File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse137.py

from flask import Flask, request, jsonify
import hashlib
import re
from functools import wraps

app = Flask(__name__)

# Example data - in production, use secure database
moderators = {
    "admin": hashlib.sha256("admin123".encode()).hexdigest(),
    "mod1": hashlib.sha256("modpass456".encode()).hexdigest()
}

valid_sessions = {
    "session123": "admin",
    "session456": "mod1"
}

def sanitize_input(input_str):
    """Sanitize input to prevent injection attacks"""
    if input_str is None:
        return ""
    return re.sub(r'[^a-zA-Z0-9]', '', input_str)

def hash_password(password):
    """Hash password using SHA-256"""
    return hashlib.sha256(password.encode()).hexdigest()

def auth_mod(username, password, session):
    """Authenticate if user is a moderator"""
    try:
        # Sanitize inputs
        username = sanitize_input(username)
        session = sanitize_input(session)
        
        if not username or not password or not session:
            return False
        
        # Verify session
        session_user = valid_sessions.get(session)
        if session_user is None or session_user != username:
            return False
        
        # Verify credentials
        stored_hash = moderators.get(username)
        if stored_hash is None:
            return False
        
        input_hash = hash_password(password)
        return stored_hash == input_hash
        
    except Exception:
        return False

@app.route('/check_mod', methods=['GET'])
def check_mod():
    """Endpoint to check moderator status"""
    # Get parameters from URL
    username = request.args.get('username', '')
    password = request.args.get('password', '')
    session = request.args.get('session', '')
    
    # Call auth_mod function
    is_moderator = auth_mod(username, password, session)
    
    if is_moderator:
        return jsonify({"status": "success", "message": "Moderator access granted"}), 200
    else:
        return jsonify({"status": "error", "message": "Access denied"}), 403

if __name__ == '__main__':
    # Test cases
    print("Test Case 1: Valid moderator admin")
    print(auth_mod("admin", "admin123", "session123"))
    
    print("\\nTest Case 2: Valid moderator mod1")
    print(auth_mod("mod1", "modpass456", "session456"))
    
    print("\\nTest Case 3: Invalid password")
    print(auth_mod("admin", "wrongpass", "session123"))
    
    print("\\nTest Case 4: Invalid session")
    print(auth_mod("admin", "admin123", "invalidsession"))
    
    print("\\nTest Case 5: Non-existent user")
    print(auth_mod("hacker", "password", "session999"))
    
    # Uncomment to run the web server
    # app.run(debug=False, host='127.0.0.1', port=8000, ssl_context='adhoc')

File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse137.cpp

#include <iostream>
#include <string>
#include <map>
#include <set>

class Lse137 {
public:
    // Function to check if user is a moderator
    static bool authMod(const std::string& username, const std::string& password, const std::string& session) {
        // Sample moderator credentials (in real application, this would be from a database)
        std::map<std::string, std::string> moderators;
        moderators["admin"] = "admin123";
        moderators["mod1"] = "modpass1";
        moderators["mod2"] = "modpass2";
        
        // Sample valid sessions (in real application, this would be validated from session store)
        std::set<std::string> validSessions = {"session123", "session456", "session789"};
        
        // Check if username exists, password matches, and session is valid
        auto it = moderators.find(username);
        if (it != moderators.end() && 
            it->second == password && 
            validSessions.find(session) != validSessions.end()) {
            return true;
        }
        return false;
    }
    
    // Function that would be called when visiting /check_mod URL
    static bool checkMod(const std::string& username, const std::string& password, const std::string& session) {
        return authMod(username, password, session);
    }
};

int main() {
    // Test case 1: Valid moderator credentials and session
    std::cout << "Test 1: " << Lse137::checkMod("admin", "admin123", "session123") << std::endl;
    
    // Test case 2: Valid moderator but invalid session
    std::cout << "Test 2: " << Lse137::checkMod("mod1", "modpass1", "invalidsession") << std::endl;
    
    // Test case 3: Invalid username
    std::cout << "Test 3: " << Lse137::checkMod("user1", "password", "session456") << std::endl;
    
    // Test case 4: Valid username but wrong password
    std::cout << "Test 4: " << Lse137::checkMod("mod2", "wrongpass", "session789") << std::endl;
    
    // Test case 5: Another valid moderator with valid session
    std::cout << "Test 5: " << Lse137::checkMod("mod2", "modpass2", "session456") << std::endl;
    
    return 0;
}
